### PR TITLE
Implement the frontend page for recipient's data input using ppa #26

### PR DESCRIPTION
**Pull request by Ngha-Boris**
Payment Options
payment option
QR-scan
qr scan
Form
form

### CODE REVIEW

File: power-pay-frontend/src/main.tsx
Comment (Motouom) [inline power-pay-frontend/src/main.tsx:1]: Please remove unnecessary imports here

**Review by Motouom — CHANGES_REQUESTED** [review]

**Comment by Motouom** [comment]
Please also make sure that your front-end matches with the mock-up that was provided.

**Review by Motouom — COMMENTED** [review]
please also send the output of the front-end pages so that we see what it looks like
for 
- the input option page
- the qr-scan page
- input manually page
@Ngha-Boris @Blindspot22 @Tekum-Emmanuella

**Comment by Ngha-Boris** [comment]
> please also send the output of the front-end pages so that we see what it looks like for
> 
> * the input option page
> * the qr-scan page
> * input manually page
>   @Ngha-Boris @Blindspot22 @Tekum-Emmanuella

ok on it

**Comment by Motouom** [comment]
image

the QR-scan has to be in a new page and on on the same page as the input option

File: power-pay-frontend/src/components/scan_rq.tsx [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+  const handleScan = (data: string | null) => {
+    if (data) {
+      console.log('Scanned data:', data);
+    }
+  };
+
+  const handleError = (err: any) => {
+    console.error('QR scanner error:', err);
+  };
```
Comment (stephane-segning) [inline power-pay-frontend/src/components/scan_rq.tsx:29]: You should do a real handling of these cases.

File: power-pay-frontend/src/components/index.tsx
Comment (stephane-segning) [inline power-pay-frontend/src/components/index.tsx:1]: Rename this file into something more specific

File: power-pay-frontend/src/components/fake_api.tsx [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+export interface ApiResponse {
+  phone: string;
+  amount: string;
+}
+
+export interface ErrorResponse {
+  message: string;
+}
+
+const FakeApi = () => {
```
Comment (stephane-segning) [inline power-pay-frontend/src/components/fake_api.tsx:12]: Use small letters for this one

**Review by stephane-segning — CHANGES_REQUESTED** [review]
Focus on these points too please

**Comment by Tekum-Emmanuella** [comment]
I have updated the userinfo for manual input and  made it responsive.

**Comment by Tekum-Emmanuella** [comment]
the userinfo for manual input has been made responsive sir

**Comment by Blindspot22** [comment]
> I have updated the userinfo for manual input and made it responsive.

@Tekum-Emmanuella Please learn to always tag people when replying their message

**Comment by Tekum-Emmanuella** [comment]
> > I have updated the userinfo for manual input and made it responsive.
> 
> @Tekum-Emmanuella Please learn to always tag people when replying their message it alerts them more precisely
Thank you @Blindspot22

**Comment by Tekum-Emmanuella** [comment]
> Please also make sure that your front-end matches with the mock-up that was provided.

ok @Motouom

**Comment by Tekum-Emmanuella** [comment]
> Please also make sure that your front-end matches with the mock-up that was provided.

ok @Motouom

**Comment by Tekum-Emmanuella** [comment]
> > Please also make sure that your front-end matches with the mock-up that was provided.
> 
> ok @Motouom

ok @Motouom

**Comment by Tekum-Emmanuella** [comment]
@Motouom

**Review by Koufan-De-King — APPROVED** [review]
@Ngha-Boris I see the tests are now successful. As far as I can see, all good.

**Comment by Ngha-Boris** [comment]
qrscanner

**Review by Motouom — APPROVED** [review]
OKAY GOOD!

**Review by mbunwe-victor — APPROVED** [review]

**Comment by Ngha-Boris** [comment]
> Focus on these points too please

done

**Review by Ngha-Boris — COMMENTED** [review]
All changes made

File: paymentoption.tsx
Comment (stephane-segning) [inline paymentoption.tsx:1]: What is the use of this item here?

**Review by stephane-segning — CHANGES_REQUESTED** [review]

File: paymentoption.tsx
Comment (Tekum-Emmanuella) [reply paymentoption.tsx:1]: It's the user interface where he/she can send money either manually or by scanning

Comment (Tekum-Emmanuella) [reply paymentoption.tsx:1]: It's the user interface where he/she can send money either manually or by scanning @stephane-segning

File: power-pay-frontend/src/components/userinfo.tsx [hunk 3]
```diff
@@ -0,0 +1,67 @@
+import React from 'react';
+import { useFormik } from 'formik';
+import { Link } from 'react-router-dom';
+interface FormValues {
+    phone: string;
+    amount: string;
+}
+
+const UserInfo: React.FC = () => {
```
Comment (stephane-segning) [inline power-pay-frontend/src/components/userinfo.tsx:9]: Rename this to the according related component's name

File: power-pay-frontend/src/components/scan_rq.tsx [hunk 4]
```diff
@@ -0,0 +1,45 @@
+import {  useEffect, useRef as ref } from 'react';
+import { Scanner as S } from '@yudiel/react-qr-scanner'; // Importing Scanner from @yudiel/react-qr-scanner
+const Scanner: any = S;
+
+
+const QRScannerComponent  = () => {
```
Comment (stephane-segning) [inline power-pay-frontend/src/components/scan_rq.tsx:6]: You might need some props for passing data in and out

File: power-pay-frontend/src/components/route.tsx [hunk 5]
```diff
@@ -0,0 +1,23 @@
+import {
+    createBrowserRouter
+  } from "react-router-dom"
+  ;
+  import PaymentOptions from "./paymentoption";
+import UserInfo from "./userinfo";
+import QRScannerComponent
+ from "./scan_rq";
```
Comment (stephane-segning) [inline power-pay-frontend/src/components/route.tsx:8]: Format this well

File: paymentoption.tsx [hunk 6]
```diff
@@ -0,0 +1,43 @@
+import React, { useState } from 'react';
+import QRScannerComponent from './scan_rq';
+import UserInfo from './userinfo'; // Import the UserInfo component
+import { Link } from "react-router-dom";
+
+const PaymentOptions: React.FC = () => {
```
Comment (stephane-segning) [inline paymentoption.tsx:6]: This might need some props to manage the IO of this component from the outside

**Review by stephane-segning — CHANGES_REQUESTED** [review]

**Comment by Tekum-Emmanuella** [comment]
Good morning .@Mr.stephane. please all the above changes you requested for have been resolved.

**Review by stephane-segning — APPROVED** [review]

File: paymentoption.tsx
Comment (stephane-segning) [inline paymentoption.tsx:1]: Remove this component from the root

**Review by stephane-segning — CHANGES_REQUESTED** [review]

**Review by Motouom — APPROVED** [review]

**Review by Arielpetit — APPROVED** [review]

File: power-pay-frontend/src/components/paymentoption.tsx [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+        <h5 className="text-center pt-9 mb-2 text-black">Choose the option you wish to pay</h5>
+      </div>
+      <div className="pt-12">
+        <Link to={`RecipientInfo`}>
+          <button onClick={handleInputManuallyClick} className='bg-blue-950 rounded-full w-80 m-auto px-4  text-white text-lg absolute  inset-x-0 mb-20 bottom-12'>Input Manually</button>
+          <br></br>
+        </Link>
+        <div className="pt-12">
+  <Link to={`QRScannerComponent`}>
+          <button onClick={handleScanQRClick} className="bg-blue-950 rounded-full w-80 m-auto px-4  text-white text-lg absolute  inset-x-0 bottom-12">
```
Comment (stephane-segning) [inline power-pay-frontend/src/components/paymentoption.tsx:33]: Add `type` `button` to this button, because it's not for a `form` tag

**Review by stephane-segning — CHANGES_REQUESTED** [review]

**Review by stephane-segning — APPROVED** [review]

**Pull request merged**